llvm/test/CodeGen/RISCV/rvv/fixed-vectors-int-buildvec.ll

```diff
@@ -259,25 +259,40 @@ define <4 x i8> @buildvec_vid_stepn3_add3_v4i8() {
 ; CHECK-NEXT:    vsetivli zero, 4, e8, mf4, ta, ma
 ; CHECK-NEXT:    vmv.v.i v9, 3
 ; CHECK-NEXT:    vid.v v8
-; CHECK-NEXT:    li a0, -3
+; CHECK-NEXT:    li a0, 253
```

Comment:
This and the regressions below are caused by `SelectionDAG::getConstant` not sign-extending the splat value. I'll try to fix this in a separate PR

Reply:
After rebasing on top of #67419 it looks like it didn't catch this one, but #67027 should still fix it

Reply:
Talked to Luke about this offline.  My summary is that while we are generating the zero vs sign extended constant for the i8, that we believe hasAllNBitUsers handles this case.  Our theory is that the difference in the LI immediate comes down to the fact that using a different constant on the LI isn't profitable (in either direction) - i.e. no rooting transform.  This seems backed up by the fact we don't see any other test differences which *are* material.